ejb/src/main/java/org/wildfly/httpclient/ejb/HttpEJBReceiver.java

```diff
@@ -202,18 +215,15 @@ public Object getResult() throws Exception {
                                 Exception exception = null;
                                 Object returned = null;
                                 try {
-
+                                    final Map<String, Object> attachments;
                                     final Unmarshaller unmarshaller = createUnmarshaller(targetContext.getUri(), targetContext.getHttpMarshallerFactory(request));
-
-                                    unmarshaller.start(new InputStreamByteInput(input));
-                                    returned = unmarshaller.readObject();
-                                    // read the attachments
-                                    final Map<String, Object> attachments = readAttachments(unmarshaller);
-                                    // finish unmarshalling
-                                    if (unmarshaller.read() != -1) {
-                                        exception = EjbHttpClientMessages.MESSAGES.unexpectedDataInResponse();
+                                    final ByteInput in = new InputStreamByteInput(input);
+                                    try (in) {
```

Comment:
Try with resources?

Reply:
This code moved to ClientHandlers. It is fixed there in the way you suggested.